src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -5263,14 +5263,13 @@ private static Class<?>[] insertArgumentsChecks(MethodHandle target, int insCoun
      *                  or if the new method handle's type would have too many parameters
      */
     public static MethodHandle dropArguments(MethodHandle target, int pos, List<Class<?>> valueTypes) {
-        return dropArguments0(target, pos, copyTypes(valueTypes.toArray()));
+        return dropArguments(target, pos, valueTypes.toArray(new Class<?>[0]).clone(), true);
```

Comment:
Isn't this call to `clone()` unnecessary, as `valueTypes.toArray` should either return the passed empty array, or a newly created array?

Reply:
It might be a bit too paranoid in this instance (since we don't keep the array around for long), but not cloning the result of calling `toArray` on an arbitrary and possibly adversary `List` could open up for TOCTOU race bugs / attacks. The existing code was being paranoid and copying and I don't want to weaken something that could have security implications without double- and triple-checking that it's safe to do so.